openstack/messaging/v2/queues/requests.go

```diff
@@ -0,0 +1,65 @@
+package queues
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToQueueCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts specifies the queue creation parameters.
+type CreateOpts struct {
+	//The target incoming messages will be moved to when a message can’t
+	// processed successfully after meet the max claim count is met.
+	DeadLetterQueue string `json:"_dead_letter_queue,omitempty"`
+
+	// The new TTL setting for messages when moved to dead letter queue.
+	DeadLetterQueueMessageTTL int `json:"_dead_letter_queue_messages_ttl,omitempty"`
+
+	// The delay of messages defined for a queue. When the messages send to
+	// the queue, it will be delayed for some times and means it can not be
+	// claimed until the delay expired.
+	DefaultMessageDelay int `json:"_default_message_delay,omitempty"`
+
+	// The default TTL of messages defined for a queue, which will effect for
+	// any messages posted to the queue.
+	DefaultMessageTTL int `json:"_default_message_ttl" required:"true"`
+
+	// The flavor name which can tell Zaqar which storage pool will be used
+	// to create the queue.
+	Flavor string `json:"_flavor,omitempty"`
+
+	// The max number the message can be claimed.
+	MaxClaimCount int `json:"_max_claim_count,omitempty"`
+
+	// The max post size of messages defined for a queue, which will effect
+	// for any messages posted to the queue.
+	MaxMessagesPostSize int `json:"_max_messages_post_size,omitempty"`
+
+	// Description of the queue.
+	Description string `json:"description,omitempty"`
```

Comment:
I can't find where `Description` is defined in the Messaging service. I searched _all_ of the code and I can only find "description" being used for error messages or for rendering API documentation.

I have a theory about this: From looking through the Zaqar code and commit messages, _all_ of these fields are Metadata, but the ones prefixed with `_` are reserved/special metadata. 

I'm thinking that the Create call will accept _any_ field. You can verify this by adding any number of additional fields to this struct such as:

```go
Foo string `json:"foo,omitempty"`
```

And try to create a queue with `Foo` set. My guess is that "foo" will be added as some metadata of the queue.

If this turns out to be true, we should create a master field called `Metadata` (but keep the reserved metadata fields as-is) and implement it similar to Keystone's "Extra" field:

https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/users/requests.go#L103
https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/users/requests.go#L119-L125